### DISCUSSION TITLE
Pydantic boolean settings cached

### DISCUSSION THREAD

**Discussion by lsmith77** [fastapi/fastapi#6460] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
class Settings(BaseSettings):
    """Load environment variables to python objects using pydantic."""

    logging_enabled: bool = False
    logging_config_filename: str = "./logs/error.log"
    logging_config_level: str = "ERROR"
    training_data_enabled: bool = False
    api_docs_username: Optional[str]
    api_docs_password: Optional[str]
    api_docs_auth_enabled: bool = False
    testing: bool = False

    class Config:
        env_file = ".env"
```

### Description

Generally FastAPI is able to read data from the `.env` file but it is then somehow cached. Restarting the server does not re-read the `.env` values. Adding a new or renaming an setting works as expected. However, changing a value of a previously read setting has no effect. Even when removing the `env_file = “.env”` bit.

The issue seems to affect boolean values only. However, it does handle False vs. True properly, both via .env as well as via proper env variables.

### Operating System

Linux, macOS

### FastAPI Version

0.68.2

### Python Version

3.9

**Comment by wcastello** [comment]
I couldn't reproduce this. 

I used your file as a config.py, this main.py:

```
from fastapi import FastAPI
from config import Settings

app = FastAPI()
settings = Settings()

@app.get("/")
async def root():
    return settings
```

and this .env:

```
API_DOCS_USERNAME="test@notsure.dev"
TESTING=false
```

Changing either API_DOCS_USERNAME's string or TESTING boolean on my .env works upon reloading the server. The values in these settings are updated as expected.

**Comment by lsmith77** [comment]
thank you for checking. but sorry, by now I am pretty sure its caused by VSCode and forgot to close this ticket.

https://github.com/microsoft/vscode-python/issues/18203
